crypto/include/internal/ciphermode_platform.h

```diff
@@ -94,7 +94,7 @@ void AES_xts_decrypt(const unsigned char *inp, unsigned char *out, size_t len,
 #  endif
 # endif /* OPENSSL_CPUID_OBJ */
 
-# if     defined(AES_ASM) && !defined(I386_ONLY) &&      (  \
+# if     !defined(OPENSSL_NO_ASM) &&                     (  \
```

Comment:
It's better if you use `defined(AES_ASM)` than `!defined(OPENSSL_NO_ASM)`, since there's potential lack of asm support even if configured without `no-asm`

Reply:
I think that will not work, because AES_ASM is no longer defined
in all cases.
Previously we had either AES_ASM or AES_ASM + BSAES_ASM + VPAES_ASM
or ASE_ASM + VPASE_ASM
but now we may also have AES_VPAES only.
But even without VPAES and BSAES we could still have aesni_encrypt/decrypt.
I did not remove aesni, that is constant time, but it needs a AESNI_CAPAPLE in
the ia32cap_P